File: Frameworks/RSCore/RSCore/AppKit/NSAppleEventDescriptor+RSCore.m
//
//  NSAppleEventDescriptor+RSCore.m
//  RSCore
//
//  Created by Brent Simmons on 1/15/18.
//  Copyright © 2018 Ranchero Software, LLC. All rights reserved.
//

#import "NSAppleEventDescriptor+RSCore.h"

@implementation NSAppleEventDescriptor (RSCore)

+ (NSAppleEventDescriptor * _Nullable)descriptorWithRunningApplication:(NSRunningApplication *)runningApplication {

	pid_t processIdentifier = runningApplication.processIdentifier;
	if (processIdentifier == -1) {
		return nil;
	}

	return [NSAppleEventDescriptor descriptorWithProcessIdentifier:processIdentifier];
}

@end

File: RSCore/Sources/RSCore/AppKit/NSAppleEventDescriptor+RSCore.swift
//
//  NSAppleEventDescriptor+RSCore.swift
//  RSCore
//
//  Created by Nate Weaver on 2020-01-02.
//  Copyright © 2020 Ranchero Software, LLC. All rights reserved.
//
#if os(macOS)
import AppKit

public extension NSAppleEventDescriptor {

	/// An NSAppleEventDescriptor describing a running application.
	///
	/// - Parameter runningApplication: A running application to associate with the descriptor.
	///
	/// - Returns: An instance of `NSAppleEventDescriptor` that refers to the running application,
	///   or `nil` if the running application has no process ID.
	convenience init?(runningApplication: NSRunningApplication) {

		let pid = runningApplication.processIdentifier
		if pid == -1 {
			return nil
		}

		self.init(processIdentifier: pid)
	}

}
#endif
